file_catalog/mongo.py

```diff
@@ -40,7 +40,6 @@ def __init__(self, host=None, port=None, authSource=None, username=None, passwor
         self.client.files.create_index([('logical_name',pymongo.HASHED)], background=True)
```

Comment:
also we have two logical_name indexes, what's up with that?

Reply:
The HASHED index is a constant time lookup, so if you have the full name it is much faster.  But it doesn't work if you don't have the full name, thus a fallback to the range-based lookup.